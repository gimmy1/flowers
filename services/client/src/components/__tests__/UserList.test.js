import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import UsersList from '../UsersList';

const users = [
    {active: true, email: "gamal@gamal.com", id: 1, username: "gamal", admin: false},
    {active: true, email: "gimmy@mgimmy.com", id: 2, username: "gimmy", admin: false}
]

test('UsersList renders properly', () => {
    const wrapper = shallow(<UsersList users={users}/>);
    const element = wrapper.find('h4')
    const td = wrapper.find('tbody > tr > td')
    console.log(td.length)
    expect(td.length).toBe(10)
    // expect(wrapper.find('h1').get(0).props.children).toBe('All Users')
    // // expect(element.length).toBe(2)
    // expect(element.get(0).props.children).toBe('gamaZ')
})

test('UsersList renders a snapshot properly', () => {
    const tree = renderer.create(<UsersList users={users}/>).toJSON()
    expect(tree).toMatchSnapshot();
});