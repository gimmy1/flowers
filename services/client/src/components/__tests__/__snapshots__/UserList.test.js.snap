// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot properly 1`] = `
<div>
  <h1
    className="title is-1"
  >
    All Users
  </h1>
  <hr />
  <br />
  <table
    className="table is-hoverable is-fullwidth"
  >
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Username
        </th>
        <th>
          Email
        </th>
        <th>
          Active
        </th>
        <th>
          Admin
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          gamal
        </td>
        <td>
          gamal@gamal.com
        </td>
        <td />
        <td />
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          gimmy
        </td>
        <td>
          gimmy@mgimmy.com
        </td>
        <td />
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
